cpp/src/arrow/compute/kernels/scalar_cast_internal.cc

```diff
@@ -281,6 +279,8 @@ void AddZeroCopyCast(Type::type in_type_id, InputType in_type, OutputType out_ty
 }
 
 static bool CanCastFromDictionary(Type::type type_id) {
+  /// TODO(GH-43010): add is_binary_view_like() here once array_take
```

Comment:
aha I also need this in Parquet, could we just add it here or add a separate pr?